sagemaker/launch/launch_utils.py

```diff
@@ -0,0 +1,240 @@
+    assert os.path.isfile(path_to_yaml), f'The model configuration YAML file, {path_to_yaml}, \
+                                           should be a file path, but got a folder.'
+    assert os.path.exists(path_to_json), f'The graph metadata JSON file, {path_to_json}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_json), f'The graph metadata JSON file, {path_to_json}, should \
+                                           be a file path, but got a folder.'
+    # TODO(JianZhang): If entry point file requires modules out of GraphStorm, we need to check
+    #                  and support a list of strings (multiple files) and a folder (multiple files
+    #                  in a folder).
+    assert os.path.exists(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, does \
+                                           not exist.'
+    assert os.path.isfile(path_to_entry), f'The SageMaker entry point file, {path_to_entry}, \
+                                           should be a file path, but got a folder.'
+
+    # create the output folder if not exist.
+    if os.path.exists(output_path):
+        assert os.path.isdir(output_path), 'Output path should be a folder name, but got a ' + \
+                                           f'file {output_path}.'
+    else:
+        os.makedirs(output_path, exist_ok=True)
+    os.makedirs(os.path.join(output_path, 'code'), exist_ok=True)
+
+    # copy the artifacts files to output_path
+    shutil.copy(path_to_entry, os.path.join(output_path, 'code', os.path.basename(path_to_entry)))
+    if output_path != os.path.dirname(path_to_model):
+        shutil.copy(path_to_model, os.path.join(output_path, os.path.basename(path_to_model)))
+    if output_path != os.path.dirname(path_to_yaml):
+        shutil.copy(path_to_yaml, os.path.join(output_path, os.path.basename(path_to_yaml)))
+    if output_path != os.path.dirname(path_to_json):
+        shutil.copy(path_to_json, os.path.join(output_path, os.path.basename(path_to_json)))
+
+    output_file = os.path.join(output_path, output_tarfile_name + '.tar.gz')
+    with tarfile.open(output_file, 'w:gz') as tar:
+        tar.add(output_path, arcname='')
+
+    return output_file
+
+
+def parse_s3_url(s3_url):
+    """ Parse the given S3 url and checkif it is a valid S3 url.
+
+    Parameters:
+    -----------
+    s3_url: str
+        The given S3 url string. Should start with "https://" or "s3://". If not, will raise
+        an assertion error.
+    
+    Returns:
+    --------
+    bucket_name: str
+        The name of S3 bucket in the url.
+    key: str
+        The S3 object in the url.
+    """
+    parsed_url = urlparse(s3_url)
+    assert parsed_url.scheme in ['s3', 'https'], (f'Incorrect S3 url was given {s3_url}')
+
+    bucket_name = parsed_url.netloc
+    key = parsed_url.path.lstrip('/')
+    return bucket_name, key
+
+
+def extract_ecr_region(ecr_uri):
```

Comment:
What do callers of this function do when it returns None? Maybe we should be raising an error instead?

Reply:
This function tries to extract region from the `ecr_uri` if possible. Follow regex logic, if not match anything, return none. The caller will use the return value on its own logic, i.e., compare this region value to input argument to ensure they match. Returning a `None` works. 

So, prefer to keep the current logic.